include/ProjectVersion.h

```diff
@@ -28,29 +29,102 @@
+		m_build(s.section( '.', 2 ).section( '-', 1 )),
+		m_compareType(c)
 	{
 	}
 
-	static int compare( const ProjectVersion & _v1,
-						const ProjectVersion & _v2 );
+	static int compare(const ProjectVersion & v1, const ProjectVersion & v2);
+
+	const int getMajor() const { return m_major; }
+	const int getMinor() const { return m_minor; }
+	const int getRelease() const { return m_release; }
+	const QString getBuild() const { return m_build; }
+	const CompareType getCompareType() const { return m_compareType; }
+	
 
+private:
+	const int m_major;
+	const int m_minor;
+	const int m_release;
+	const QString m_build;
+	const CompareType m_compareType;
 } ;
 
 
 
 
-inline bool operator<( const ProjectVersion & _v1, const char * _str )
+inline int compStr(const ProjectVersion & v1, const char * v2)
+{
+	return ProjectVersion::compare(v1, ProjectVersion(v2));
+}
+
+
+
+inline int compStr(const ProjectVersion & v1, const QString & v2)
 {
-	return ProjectVersion::compare( _v1, ProjectVersion( _str ) ) < 0;
+	return ProjectVersion::compare(v1, ProjectVersion(v2));
 }
 
 
 
+/*
+ * ProjectVersion v. char[]
+ */
+inline bool operator<(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) < 0; }
+inline bool operator>(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) > 0; }
+inline bool operator<=(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) <= 0; }
+inline bool operator>=(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) >= 0; }
+inline bool operator==(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) == 0; }
+inline bool operator!=(const ProjectVersion & v1, const char * v2) { return compStr(v1, v2) != 0; }
+
+inline bool operator<(const char * v1, const ProjectVersion & v2) { return 0 < compStr(v2, v1); }
+inline bool operator>(const char * v1, const ProjectVersion & v2) { return 0 > compStr(v2, v1); }
+inline bool operator<=(const char * v1, const ProjectVersion & v2) { return 0 <= compStr(v2, v1); }
+inline bool operator>=(const char * v1, const ProjectVersion & v2) { return 0 >= compStr(v2, v1); }
+inline bool operator==(const char * v1, const ProjectVersion & v2) { return 0 == compStr(v2, v1); }
+inline bool operator!=(const char * v1, const ProjectVersion & v2) { return 0 != compStr(v2, v1); }
+
+/*
+ * ProjectVersion v. QString
+ */
+inline bool operator<(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) < 0; }
+inline bool operator>(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) > 0; }
+inline bool operator<=(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) <= 0; }
+inline bool operator>=(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) >= 0; }
+inline bool operator==(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) == 0; }
+inline bool operator!=(const ProjectVersion & v1, const QString & v2) { return compStr(v1, v2) != 0; }
```

Comment:
> I'd make operator>, operator<, … always compare on build level (on the finest level). Other levels would only be supported by calling something like ProjectVersion::compare("1.1.0", "1.1.5", CompareType::Minor).

Yeah... I thought of that, we can do that now by renaming compStr() compare() and making it public, then adding that param.   This usage was based on code readability, and that is just preference.

You could argue the class is more reusable with the compare() statement forced for non-build compares, but we don't really have many use-cases for a ProjectVersion reuse and the class is light weight enough IMO.

Subclassing would actually help in this regard too... but that's a lot of work for a verson comparison. :O)